src/MonitorProfiler/Communication/IpcCommServer.cpp

```diff
@@ -0,0 +1,141 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "IpcCommClient.h"
+#include "IpcCommServer.h"
+
+IpcCommServer::IpcCommServer() : _shutdown(false)
+{
+}
+
+IpcCommServer::~IpcCommServer()
+{
+    _domainSocket.~SocketWrapper();
```

Comment:
Why does the dtor have to be explicitly called? Is it because you explicitly implement the dtor for `IpcCommServer`?

Reply:
I want to make sure the inner socket is closed first before I attempt any path cleanup.